File: testbed_ruby/test/attributes/defaulted_test.rb

require 'test/unit'
require 'src/door_b'
require 'src/door_d'
require 'src/door_f'

module CruiseAttributesTest
class DefaultedTest < Test::Unit::TestCase

  def test_defaulted
    door = DoorD.new
    
    assert_equal("1",door.get_id)
    door.set_id "2"
    assert_equal("2",door.get_id)
    assert_equal(true, door.reset_id)
    assert_equal("1",door.get_id)
    
    assert_equal(2,door.get_intId)
    door.set_intId 3
    assert_equal(3,door.get_intId)
    assert_equal(true, door.reset_intId)
    assert_equal(2,door.get_intId)

    assert_equal(3.4,door.get_doubleId,0.01)
    door.set_doubleId 33.44
    assert_equal(33.44,door.get_doubleId,0.01)
    assert_equal(true, door.reset_doubleId)
    assert_equal(3.4,door.get_doubleId,0.01)

    assert_equal(Date.parse("1978-12-05"),door.get_dateId)
    door.set_dateId(Date.parse("1978-01-02"))
    assert_equal(Date.parse("1978-01-02"),door.get_dateId)
    assert_equal(true, door.reset_dateId)
    assert_equal(Date.parse("1978-12-05"),door.get_dateId)

    assert_equal(Time.parse("10:11:15"),door.get_timeId)
    door.set_timeId(Time.parse("13:14:15"))
    assert_equal(Time.parse("13:14:15"),door.get_timeId)
    assert_equal(true, door.reset_timeId)
    assert_equal(Time.parse("10:11:15"),door.get_timeId)

    assert_equal(false,door.get_booleanId)
    door.set_booleanId(true)
    assert_equal(true,door.get_booleanId)
    assert_equal(true, door.reset_booleanId)
    assert_equal(false,door.get_booleanId)

    assert_equal(5,door.get_doorId.get_id)
    door.set_doorId(DoorB.new(6))
    assert_equal(6,door.get_doorId.get_id)
    assert_equal(true, door.reset_doorId)
    assert_equal(5,door.get_doorId.get_id)
  end
end
end

File: testbed/test/cruise/attributes/test/DefaultedTest.java
package cruise.attributes.test;

import org.junit.*;
import java.sql.*;

public class DefaultedTest
{
  
  @Test
  public void defaulted()
  {
    DoorD door = new DoorD();
    
    Assert.assertEquals("1",door.getId());
    Assert.assertEquals(true, door.setId("2"));
    Assert.assertEquals("2",door.getId());
    Assert.assertEquals(true, door.resetId());
    Assert.assertEquals("1",door.getId());
    
    Assert.assertEquals(2,door.getIntId());
    Assert.assertEquals(true, door.setIntId(3));
    Assert.assertEquals(3,door.getIntId());
    Assert.assertEquals(true, door.resetIntId());
    Assert.assertEquals(2,door.getIntId());

    Assert.assertEquals(3.4,door.getDoubleId(),0.01);
    Assert.assertEquals(true, door.setDoubleId(33.44));
    Assert.assertEquals(33.44,door.getDoubleId(),0.01);
    Assert.assertEquals(true, door.resetDoubleId());
    Assert.assertEquals(3.4,door.getDoubleId(),0.01);

    Assert.assertEquals(new Date(1234),door.getDateId());
    Assert.assertEquals(true, door.setDateId(new Date(4321)));
    Assert.assertEquals(new Date(4321),door.getDateId());
    Assert.assertEquals(true, door.resetDateId());
    Assert.assertEquals(new Date(1234),door.getDateId());

    Assert.assertEquals(new Time(1235),door.getTimeId());
    Assert.assertEquals(true, door.setTimeId(new Time(5321)));
    Assert.assertEquals(new Time(5321),door.getTimeId());
    Assert.assertEquals(true, door.resetTimeId());
    Assert.assertEquals(new Time(1235),door.getTimeId());

    Assert.assertEquals(false,door.getBooleanId());
    Assert.assertEquals(true, door.setBooleanId(true));
    Assert.assertEquals(true,door.getBooleanId());
    Assert.assertEquals(true, door.resetBooleanId());
    Assert.assertEquals(false,door.getBooleanId());

    Assert.assertEquals(new DoorB(5),door.getDoorId());
    Assert.assertEquals(true, door.setDoorId(new DoorB(6)));
    Assert.assertEquals(new DoorB(6),door.getDoorId());
    Assert.assertEquals(true, door.resetDoorId());
    Assert.assertEquals(new DoorB(5),door.getDoorId());
  }

}
